sklearn/datasets/openml.py

```diff
@@ -0,0 +1,415 @@
+        if target_column_name not in features_dict:
+            raise KeyError('Could not find target_column_name={}')
+        if features_dict[target_column_name]['data_type'] == "numeric":
+            found_types.add(np.float64)
+        else:
+            found_types.add(object)
+
+        # note: we compare to a string, not boolean
+        if features_dict[target_column_name]['is_ignore'] == 'true':
+            warn('target_column_name={} has flag is_ignore.'.format(
+                target_column_name))
+        if features_dict[target_column_name]['is_row_identifier'] == 'true':
+            warn('target_column_name={} has flag is_row_identifier.'.format(
+                target_column_name))
+    if len(found_types) != 1:
+        raise ValueError('Can only handle homogeneous multi-target datasets, '
+                         'i.e., all targets are either numeric or '
+                         'categorical.')
+    return list(found_types)[0]
+
+
+def fetch_openml(name=None, version='active', data_id=None, data_home=None,
+                 target_column_name='default-target', cache=True):
+    """Fetch dataset from openml by name or dataset id.
+
+    Datasets are uniquely identified by either an integer ID or by a
+    combination of name and version (i.e. there might be multiple
+    versions of the 'iris' dataset). Please give either name or id
+    (not both). In case a name is given, a version can also be
+    provided.
+
+    Parameters
+    ----------
+    name : str or None
+        String identifier of the dataset. Note that OpenML can have multiple
+        datasets with the same name.
+
+    version : integer or 'active', default='active'
+        Version of the dataset. Can only be provided if also ``name`` is given.
+        If 'active' the oldest version that's still active is used.
+
+    data_id : int or None
+        OpenML ID of the dataset. The most specific way of retrieving a
+        dataset. If data_id is not given, name (and potential version) are
+        used to obtain a dataset.
+
+    data_home : string or None, default None
+        Specify another download and cache folder for the data sets. By default
+        all scikit-learn data is stored in '~/scikit_learn_data' subfolders.
+
+    target_column_name : string, list or None, default 'default-target'
+        Specify the column name in the data to use as target. If
+        'default-target', the standard target column a stored on the server
+        is used. If ``None``, all columns are returned as data and the
+        target is ``None``. If list (of strings), all columns with these names
+        are returned as multi-target (Note: not all scikit-learn classifiers
+        can handle all types of multi-output combinations)
+
+    cache : boolean, default=True
+        Whether to cache downloaded datasets using joblib.
+
+    Returns
+    -------
+
+    data : Bunch
+        Dictionary-like object, the interesting attributes are:
```

Comment:
I think we need to give more specification of each field, e.g. the type (and dtype) of `data`. Use a ReST definition list.

Reply:
Should also mention how missing values are returned.

Reply:
Can you clarify what you mean with a `ReST definition list`? I tried to improve this a bit (available next push)

Reply:
Agreed